keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -142,6 +143,28 @@ func GetOSInterface() types.OSTypeInstaller {
 	}
 }
 
+// RunningModuleV2 identifies cloudcore/edgecore running or not.
+// only used for cloudcore container install and edgecore binary install
+func RunningModuleV2(opt *types.ResetOptions) (types.ModuleRunning, error) {
+	osType := GetOSInterface()
+	cloudCoreRunning, err := IsCloudcoreContainerRunning(constants.SystemNamespace, opt.Kubeconfig)
+	if err != nil {
+		return types.NoneRunning, err
+	}
+	if cloudCoreRunning {
+		return types.KubeEdgeCloudRunning, nil
+	}
+
+	edgeCoreRunning, err := osType.IsKubeEdgeProcessRunning(KubeEdgeBinaryName)
+	if edgeCoreRunning {
+		return types.KubeEdgeEdgeRunning, nil
+	} else if err != nil {
+		return types.NoneRunning, err
+	}
+
+	return types.NoneRunning, nil
```

Comment:
If it is not in the cloud, it defaults to the edge.
HDYT?

Reply:
if edgecore also is not running, we need do nothing.So it's better to distinguish these. :)